srml/assets/src/lib.rs

```diff
@@ -0,0 +1,228 @@
+#[cfg(test)]
+extern crate substrate_primitives;
+
+// Needed for deriving `Serialize` and `Deserialize` for various types.
+// We only implement the serde traits for std builds - they're unneeded
+// in the wasm runtime.
+#[cfg(feature = "std")]
+#[macro_use]
+extern crate serde_derive;
+
+// Needed for deriving `Encode` and `Decode` for `RawEvent`.
+#[macro_use]
+extern crate parity_codec_derive;
+extern crate parity_codec as codec;
+
+// Needed for type-safe access to storage DB.
+#[macro_use]
+extern crate srml_support as runtime_support;
+
+// Needed for various traits. In our case, `OnFinalise`.
+extern crate sr_primitives as primitives;
+// `system` module provides us with all sorts of useful stuff and macros
+// depend on it being around.
+extern crate srml_system as system;
+
+use primitives::traits::OnFinalise;
+use runtime_support::{StorageValue, StorageMap, dispatch::Result, Parameter};
+use primitives::traits::{Member, SimpleArithmetic, Zero};
+use system::ensure_signed;
+
+pub trait Trait: system::Trait {
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+	/// The units in which we record balances.
+	type Balance: Member + Parameter + SimpleArithmetic + Default + Copy;
+}
+
+type AssetId = u32;
+
+decl_module! {
+	// Simple declaration of the `Module` type. Lets the macro know what its working on.
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		/// Issue a new class of fungible assets. There are, and will only ever be, `total`
+		/// such assets and they'll all belong to the `origin` initially. It will have an
+		/// identifier `AssetId` instance: this will be specified in the `Issued` event.
+		fn issue(origin, total: T::Balance) -> Result;
+
+		/// Move some assets from one holder to another.
+		fn transfer(origin, id: AssetId, target: T::AccountId, total: T::Balance) -> Result;
+
+		/// Destroy any assets of `id` owned by `origin`.
+		fn destroy(origin, id: AssetId) -> Result;
+	}
+}
+
+/// An event in this module. Events are simple means of reporting specific conditions and
+/// circumstances that have happened that users, Dapps and/or chain explorers would find
+/// interesting and otherwise difficult to detect.
+decl_event!(
+	pub enum Event<T> where <T as system::Trait>::AccountId, <T as Trait>::Balance {
+		/// Some assets were issued.
+		Issued(AssetId, AccountId, Balance),
+		/// Some assets were transfered.
+		Transfered(AssetId, AccountId, AccountId, Balance),
+		/// Some assets were destroyed.
+		Destroyed(AssetId, AccountId, Balance),
+	}
+);
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Assets {
+		/// The number of units of assets held by any given account.
+		Balances: map (AssetId, T::AccountId) => T::Balance;
+		/// The next asset identifier up for grabs.
+		NextAssetId get(next_asset_id): AssetId;
+
+		// Required for compailation @bkchr needs fixing! :)
+		Foo get(foo) config(): T::Balance;
```

Comment:
@bkchr this needs addressing :)

Reply:
I know, I'm already on it and have working implementation. Just need some more tweaks. Will push it to this branch :)